dataset/operations.cpp

```diff
@@ -49,7 +49,7 @@ DataArray copy(const DataArray &array, const AttrPolicy attrPolicy) {
 
 /// Return a deep copy of a Dataset.
 Dataset copy(const Dataset &dataset, const AttrPolicy attrPolicy) {
-  Dataset out({}, copy(dataset.coords()));
+  Dataset out{};
   for (const auto &item : dataset)
     out.setData(item.name(), copy(item, attrPolicy));
```

Comment:
This will drop coords of datasets without items. Need to insert coords after data?